pkg/controller/deployment/sync.go

```diff
@@ -510,15 +510,26 @@ func (dc *DeploymentController) updateDeploymentStatus(allRSs []*extensions.Repl
 
 // isScalingEvent checks whether the provided deployment has been updated with a scaling event
 // by looking at the desired-replicas annotation in the active replica sets of the deployment.
-func (dc *DeploymentController) isScalingEvent(d *extensions.Deployment) bool {
+func (dc *DeploymentController) isScalingEvent(d *extensions.Deployment) (bool, error) {
 	newRS, oldRSs, err := dc.getAllReplicaSetsAndSyncRevision(d, false)
 	if err != nil {
-		return false
+		return false, err
 	}
 	// If there is no new replica set matching this deployment and the deployment isn't paused
 	// then there is a new rollout that waits to happen
 	if newRS == nil && !d.Spec.Paused {
```

Comment:
@pwittrock then in case of compression we cannot be sure if (C) is (A) or (B). I think it still makes sense to let the rollout process handle the scaling part in such a case instead of picking (A) or (B). We may want to document this behavior.